src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,699 @@
+    /// d = (a - b)  / a
+    ///
+    /// @param venue to query
+    /// @param totalPcv to measure venue against
+    function getVenueBalance(
+        address venue,
+        uint256 totalPcv
+    ) public view returns (int256) {
+        uint256 venueDepositedVcon = venueVconDeposited[venue];
+        uint256 venueBalance = pcvOracle().getVenueBalance(venue);
+        uint256 cachedVconStaked = vconStaked;
+
+        /// 0 checks as any 0 denominator will cause a revert
+        if (
+            (venueDepositedVcon == 0 && venueBalance == 0) ||
+            cachedVconStaked == 0 ||
+            totalPcv == 0
+        ) {
+            return 0; /// perfectly balanced at 0 PCV or 0 VCON staked
+        }
+
+        /// Step 1.
+        /// find out actual ratio of VCON in a given venue based on total VCON staked
+        uint256 venueDepositedVconRatio = (venueDepositedVcon *
+            Constants.ETH_GRANULARITY) / cachedVconStaked;
+
+        /// perfectly balanced
+        if (venueDepositedVconRatio == 0 && venueBalance == 0) {
+            return 0;
+        }
+
+        if (venueDepositedVconRatio == 0) {
+            /// add this 0 check because deviation divides by a and would cause a revert
+            /// replicate step 3 if no VCON is deposited by comparing pcv to venue balance
+            return
+                DeviationWeiGranularity.calculateDeviation(
+                    totalPcv.toInt256(),
+                    venueBalance.toInt256()
+                );
+        }
+
+        /// Step 2.
+        /// get expected pcv amount
+        uint256 expectedPcvAmount = (venueDepositedVconRatio * totalPcv) /
+            Constants.ETH_GRANULARITY;
+
+        /// perfectly balanced = (expectedPcvAmount - venueBalance) * 1e18 / expectedPcvAmount = 0
+
+        /// Step 3.
+        /// if venue deposited VCON, return the ratio between expected pcv vs actual pcv
+        return
+            -DeviationWeiGranularity.calculateDeviation(
+                expectedPcvAmount.toInt256(),
+                venueBalance.toInt256()
+            );
+    }
+
+    /// @param venue to figure out total pro rata pcv
+    /// @param amountVcon to find total amount of pro rata pcv
+    /// @return the pro rata pcv controll  ed in the given venue based on the amount of VCON
+    function getProRataPCVAmounts(
+        address venue,
+        uint256 amountVcon
+    ) public view returns (uint256) {
+        uint256 venuePcv = IPCVDeposit(venue).balance();
+        uint256 cachedVconStaked = venueVconDeposited[venue];
+
+        console.log("cachedVconStaked: ", cachedVconStaked);
+
+        /// 0 checks as any 0 denominator will cause a revert
+        if (cachedVconStaked == 0) {
+            return 0; /// perfectly balanced at 0 PCV or 0 VCON staked
+        }
+
+        uint256 proRataPcv = (amountVcon * venuePcv) / cachedVconStaked;
```

Comment:
+ 1 to round up, to round in favor of the protocol

Reply:
went through things on a more in depth level, and removed requirement to round up. the newly added check around unrealized losses means this rounding up is not needed.